com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -0,0 +1,1151 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Profiling;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// The NGO connection manager handles:
+    /// - Client Connections
+    /// - Client Approval
+    /// - Processing <see cref="NetworkEvent"/>s.
+    /// - Client Disconnection
+    /// - MessagingSystem updates
+    /// </summary>
+    // TODO 2023-Q2: Discuss what kind of public API exposure we want for this
+    public class NetworkConnectionManager : INetworkUpdateSystem
```

Comment:
So thinking back to the original document for some of this was to have NetworkManager be the one that would call into the various components registered to it. Making this INetworkUpdateSystem sounds like it'll be registered with NetworkUpdateLoop instead. Is that just an intermediate step?

Main reason this concerns me as an intermediate step is that, when we create a different interface to be used for NetworkManager (since the way NetworkManager processes updates is a little different with the inclusion of dependencies within each stage), that'd be a breaking change from this. So if this goes into develop as it is, we'd be blocked from releasing anything until we get the next step done.

Reply:
Actually a good point... this should have had an internal INetworkUpdateSystem implementation that gets instantiated and hides the public NetworkUpdate method within NetworkConnectionManager itself.  That will avoid any future changes where perhaps the update is handled elsewhere (i.e. removing the implementation won't be considered a breaking change).
Regarding the order in which things are updated, they are still updated in the same order...just not all within NetworkManager.

Reply:
Since we're not removing `NetworkManager.NetworkUpdate(...)` method yet, and `NetworkManager` is still the service locator for network systems, we can just simply call update on network systems (i.e. `NetworkConnectionManager`) from outside, from `NetworkManager.NetworkUpdate(...)` up until we make breaking changes.